perllib/FixMyStreet/Cobrand/Merton/Waste.pm

```diff
@@ -376,6 +376,34 @@ sub waste_post_report_creation {
     }
 }
 
+sub check_ggw_transfer_applicable {
+    my ($self, $old_address) = @_;
+
+    my $error;
+    my $expiry;
+
+    # Check new address doesn't have a ggw subscription
+    $error = 'current' if $self->garden_current_subscription;
+
+    # Check that the old address has a ggw subscription and it's not
+    # in its expiry period
+    my $details = $self->look_up_property($old_address);
+    my $old_services = $self->{api_serviceunits};
+
+    # Reset things as this is still the property we're actually on
+    $self->look_up_property($self->{c}->stash->{property}{id});
+
+    my ($old_garden) = grep { $_->{ServiceId} eq '409' } @$old_services;
+    $old_garden->{transfer_uprn} = $details->{uprn};
+
+    $error = 'no_previous' unless $error || ($old_garden && $old_garden->{ServiceTasks});
+    my $subscription_enddate = $old_garden->{ServiceTasks}->{ServiceTask}->{ServiceTaskSchedules}->{ServiceTaskSchedule}[0]->{EndDate}->{DateTime} unless $error;
```

Comment:
You cannot assume this structure, Echo has some strange data in it with multiple services overlapping or what have you. Probably not the case for garden I hope, but better safe and to be consistent with other uses. I think you can call `garden_current_service_from_service_units($old_services)` to get the current service task, and then call `_parse_schedules(that servicetask)` to get something containing `->{end_date}` that matches what is displayed to the user.

Reply:
Also, if we look this up here, can we store it on $old_garden to not need to look it up again, like you've done with transfer_uprn